install-and-configure/install/multi-cluster/federated-etl/thanos-migration-guide.md

```diff
@@ -0,0 +1,154 @@
+```
+
+### Step 3: Validate that an `/etl` directory is present in the object store
+
+There should be ETL data present in the following directories. CloudCosts will only have ETL data if cloud integration is enabled.
+
+* `/etl/bingen/allocations`
+* `/etl/bingen/assets/`
+* `/cloudcosts/`
+
+
+### Step 4: Create a new secret for the federated store
+
+This will point to the existing thanos object store or the new one created in Step 1.
+
+{% hint style="warning" %}
+The name of the .yaml file used to create the secret *must* be named _federated-store.yaml_ or Aggregator will not start.
+{% endhint %}
+
+
+```
+kubectl create secret generic federated-store --from-file=federated-store.yaml -n kubecost
+```
+
+### Step 5: Set Aggregator and ETL Utils in *values.yaml* on Primary Kubecost Instance.
+
+{% hint style="warning" %}
+*Do not* disable Thanos during this step. This will not negatively impact the source data. Thanos will be disabled in a later step.
+{% endhint %}
+
+* [Enable Aggregator (primary kubecost instance only)](https://docs.kubecost.com/install-and-configure/install/multi-cluster/federated-etl/aggregator). 
+
+Note:  
+
+* ETL Utils
+```
+etlUtils:
+  thanosSourceBucketSecret: kubecost-thanos
+  affinity: {}
+  enabled: true
+  env:
+    LOG_LEVEL: debug
+  fullImageName: gcr.io/kubecost1/cost-model-etl-utils:0.2
+  nodeSelector:
+    kubernetes.io/arch: amd64
+  resources: {}
+  tolerations: {}
+kubecostModel:
+  federatedStorageConfigSecret: kubecost-thanos
+```
+
+### Step 6: Apply the changes and wait for data to populate in the UI. 
+
+Ensure all pods and containers are running:
+
+ ```
+kubecost-cost-analyzer-685fd8f677-k652h        4/4     Running   2 (12m ago)   3h2m
+kubecost-forecasting-6cf4dd7b98-7z8f5          1/1     Running   0             66m
+kubecost-etl-utils-6cdd489596-5dl75           1/1     Running   0          6d20h
+```
+
+This can take some time depending on how many unique containers are running in the environment and how much ETL data is in the object store. A couple importants steps are happening in the background which take time to complete.
+
+* The ETL Utils image is building the directory structure in the object store needed by Aggregator to pull the ETL data. 
+* SQL tables are building.
+
+Ensure all data loads into the UI before moving onto step 7.
+
+### Step 7: Remove Thanos sidecar from Secondary Clusters.
+
+* Use the same federated-store secret created in step 4 and add the following to the values.yaml for the secondary clusters:
+
+```
+federatedETL:
+  useExistingS3Config: false
+  primaryCluster: false
+  federatedCluster: true
+kubecostModel:
+  containerStatsEnabled: true
+  federatedStorageConfigSecret: federated-store
+  warmCache: false
+  warmSavingsCache: false
+```
+
+* Remove the [thanos manifest](https://raw.githubusercontent.com/kubecost/cost-analyzer-helm-chart/v1.108.1/cost-analyzer/values-thanos.yaml)
+
+
+### Step 8: Remove Thanos Configuration from Primary.
```

Comment:
Depending on how the user is applying changes to their values-thanos.yaml, will they need to perform a `helm` upgrade at the end to validate the changes?

Reply:
That's one method - the common use case we see with our customers is through CI/CD (e.g. argo, flux, etc). If we want to standardize this per our other examples throughout our documentation we can and users will get the idea since helm integrates with these platforms.